sheer/indexer.py

```diff
@@ -149,8 +149,21 @@ def index_location(args, config):
                                 doc_type=processor.name,
                                 body={processor.name: processor.mapping(default_mapping)})
 
+        # Get the document iterator from the processor.
+        # This could raise a value error raised by json.loads() if there is a
+        # connection error, server error, or other error acquiring the data.
+        # This is seperate from the for loop so that we dont create/update some
+        # of the documents then error.
+        try:
+            document_iterator = enumerate(processor.documents())
+        except ValueError:
+            # If there's an error, log it, and then move on to the next content
+            # processor.
+            sys.stderr.write("error reading documents for %s" % processor.name)
+            continue
```

Comment:
I should note too that it's entirely possible we need to catch other exceptions here, or perhaps just Exception (and then use the exception object in the error it writes out). It seems like ValueError is what we'll get from the content processors we have though. They necessarily have to load JSON (since we just shove the resulting document into Elasticsearch presuming it's JSON), and `json.loads()` is where we're seeing any potential error that happens in production, and AFAIK `json.loads()` only throws JSONDecodeErrors, which is a subclass of ValueError.